src/Display/Modal/Modal.tsx

```diff
@@ -52,8 +60,11 @@ class Modal extends React.Component<Props, State> {
   }
 
   componentWillUnmount() {
-    const { onClose } = this.props;
-    document.removeEventListener('keydown', escEvent(onClose), false);
+    document.removeEventListener(
+      'keydown',
+      escEvent(this.guardedOnClose),
```

Comment:
this event listener will not be removed, because `escEvent(this.guardedOnClose)` creates another function. We need to pass the same event handler function we passed to `addEventListener` here.

Reply:
Yeah I'm not trying to fix that with this PR.

With this the event handler will still be called but it won't call `onClose` because of the check.

Reply:
@tingtingdisco so is this good?

Reply:
@westwood846 sorry for the late reply, I think the event handler should be correctly removed, it's related to your change.

Reply:
Updated!